src/Components/QuickGrid/Microsoft.AspNetCore.Components.QuickGrid/test/GridRaceConditionTest.cs

```diff
@@ -0,0 +1,144 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.AspNetCore.Components.QuickGrid;
+using Microsoft.AspNetCore.Components.Test.Helpers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.QuickGrid.Test;
+
+public class GridRaceConditionTest
+{
+    private TestRenderer _renderer = new();
+    private TaskCompletionSource _tcs = new();
+
+    public GridRaceConditionTest()
+    {
+        var testJsRuntime = new TestJsRuntime(_tcs);
+        var serviceProvider = new ServiceCollection()
+            .AddSingleton<IJSRuntime>(testJsRuntime)
+            .BuildServiceProvider();
+        _renderer = new(serviceProvider);
+    }
+
+    [Fact]
+    public async Task CanCorrectlyDisposeAsync()
+    {
+        var testComponent = new TestComponent();
+
+        var componentId = _renderer.AssignRootComponentId(testComponent);
+        _renderer.RenderRootComponent(componentId);
+        await Task.Delay(10);
```

Comment:
Will this work with only `await Task.Yield()`? In the meaning: would the test fail consistently without the change from `src/Components/QuickGrid/Microsoft.AspNetCore.Components.QuickGrid/src/QuickGrid.razor.cs` if we remove the delay? Each delay prolongs the execution of the test suite, I'm looking for a way to avoid it.